README.rst

```diff
@@ -340,6 +340,57 @@ unconventional syntax (without the usual ``@``):
     if isinstance(res, Ok):
         print(res.value)
 
+
+Do notation: syntactic sugar for a sequence of `and_then()` calls.
+Much like the equivalent in Rust or Haskell, but with different syntax.
+Instead of `x <- Ok(1)` we write `for x in Ok(1)`.
+Since the syntax is generator-based, the final result must be the first line,
+not the last.
+
+.. sourcecode:: python
+
+
+    >>> final_result: Result[float, int] = do(
+            Ok(len(x) + int(y) + 0.5)
+            for x in Ok("hello")
+            for y in Ok(True)
+        )
+
+    >>> # This is similar to
+    use do_notation::m;
+    let final_result = m! {
+        x <- Ok("hello");
+        y <- Ok(True);
+        Ok(len(x) + int(y) + 0.5)
+    };
```

Comment:
Could we slightly alter this here,

1. Perhaps word it like, `this is similar to the Rust macro library <insert library name + link>:`
2. Make the rust snippet separate and make it `.. sourcecode:: rust` so it syntax highlights correcty